sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -866,10 +861,7 @@ static int _match_to_idx(const gnrc_netif_t *netif,
     for (int i = 0; i < GNRC_NETIF_IPV6_ADDRS_NUMOF; i++) {
         unsigned match;
 
-        if ((netif->ipv6.addrs_flags[i] == 0) ||
-            ((filter != NULL) && _addr_anycast(netif, i)) ||
```

Comment:
See [RFC 6724, page 30](https://tools.ietf.org/html/rfc6724#page-30), point 3: This can be removed without alternative.